homework6/task2_oop2/project2_oop2/oop2.py

```diff
@@ -0,0 +1,149 @@
+объект - будем возвращать какой-то результат работы (HomeworkResult)
+HomeworkResult принимает объект автора задания, принимает исходное задание
+и его решение в виде строки
+Атрибуты:
+    homework - для объекта Homework, если передан не этот класс -  выкинуть
+    подходящие по смыслу исключение с сообщением:
+    'You gave a not Homework object'
+    solution - хранит решение ДЗ как строку
+    author - хранит объект Student
+    created - c точной датой и временем создания
+2. Если задание уже просрочено хотелось бы видеть исключение при do_homework,
+а не просто принт 'You are late'.
+Поднимайте исключение DeadlineError с сообщением 'You are late' вместо print.
+3. Student и Teacher имеют одинаковые по смыслу атрибуты
+(last_name, first_name) - избавиться от дублирования с помощью наследования
+4.
+Teacher
+Атрибут:
+    homework_done - структура с интерфейсом как в словаря, сюда поподают все
+    HomeworkResult после успешного прохождения check_homework
+    (нужно гаранитровать остутствие повторяющихся результатов по каждому
+    заданию), группировать по экземплярам Homework.
+    Общий для всех учителей. Вариант ипользования смотри в блоке if __main__...
+Методы:
+    check_homework - принимает экземпляр HomeworkResult и возвращает True если
+    ответ студента больше 5 символов, так же при успешной проверке добавить в
+    homework_done.
+    Если меньше 5 символов - никуда не добавлять и вернуть False.
+    reset_results - если передать экземпряр Homework - удаляет только
+    результаты этого задания из homework_done, если ничего не передавать,
+    то полностью обнулит homework_done.
+PEP8 соблюдать строго.
+Всем перечисленным выше атрибутам и методам классов сохранить названия.
+К названием остальных переменных, классов и тд. подходить ответственно -
+давать логичные подходящие имена.
+"""
+import datetime
+from collections import defaultdict
+from typing import Any
+
+
+class Homework:
+    """Create Homework object."""
+
+    def __init__(self, text: str, days_to_deadline: int) -> None:
+        self.text = text
+        self.deadline = datetime.timedelta(days=days_to_deadline)
+        self.created = datetime.datetime.now()
+
+    def is_active(self) -> bool:
+        """Check if there is time till deadline of current homework.
+
+        Return True if the deadline has not expired, False otherwise.
+        """
+        return self.created + self.deadline > datetime.datetime.now()
+
+
+class DeadlineError(Exception):
+    """Raise when homework is inactive when its deadline has already ended."""
+
+
+class Person:
+    """Create an instance of a person. Base class for Student and Teacher classes."""
+
+    def __init__(self, first_name: str, last_name: str):
+        self.first_name = first_name
+        self.last_name = last_name
+
+
+class Student(Person):
+    """Create Student object."""
+
+    def do_homework(self, home_work: Homework, result: str) -> Any:
```

Comment:
Why any?  There is only one option for return value here and it is HomeworkResult